src/pages/home/report/ContextMenu/ContextMenuActions.js

```diff
@@ -96,20 +97,28 @@ export default [
         // the `text` and `icon`
         onPress: (closePopover, {reportAction, selection}) => {
             const message = _.last(lodashGet(reportAction, 'message', [{}]));
-            const html = lodashGet(message, 'html', '');
-
-            const parser = new ExpensiMark();
-            const reportMarkdown = parser.htmlToMarkdown(html);
-
-            const text = selection || reportMarkdown;
+            const messageHtml = lodashGet(message, 'html', '');
 
             const isAttachment = _.has(reportAction, 'isAttachment')
                 ? reportAction.isAttachment
                 : ReportUtils.isReportMessageAttachment(message);
             if (!isAttachment) {
-                Clipboard.setString(text);
+                // This error indicates new code that uses the selection object incorrectly.
+                if (selection && (selection.text || !selection.html)) {
+                    throw new Error('Selection has to either exclusively bear html or be null.');
+                }
+                const resolvedSelection = selection || SelectionScraper.prepareSelection(messageHtml);
```

Comment:
Why do we have to run the messageHtml through the dom renderer? It is already custom HTML from our parser. Also, what about natives. That's why I think we don't really need to export `SelectionScraper.prepareSelection`. You can just use it in getCurrentSelection for index.js file and just create a simple object `{html: messageHtml}` here.

This way we will also eliminate the use of `BaseSelectionScraper`.